packages/react-components/src/components/ChatMessage/FluentChatMessageComponentWrapper.tsx

```diff
@@ -0,0 +1,281 @@
+            /* @conditional-compile-remove(file-sharing) */
+            fileDownloadHandler={fileDownloadHandler}
+          />
+        );
+      }
+      return <></>;
+    },
+    [
+      onActionButtonClick,
+      onRenderAvatar,
+      onRenderFileDownloadsMemo,
+      participantCount,
+      shouldOverlapAvatarAndMessage,
+      showMessageStatus,
+      userId,
+      /* @conditional-compile-remove(date-time-customization) */
+      onDisplayDateTimeString,
+      /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+      onFetchInlineAttachment,
+      /* @conditional-compile-remove(image-gallery) */
+      onInlineImageClicked,
+      /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+      inlineAttachments,
+      /* @conditional-compile-remove(mention) */
+      mentionOptions,
+      /* @conditional-compile-remove(file-sharing) */
+      fileDownloadHandler
+    ]
+  );
+
+  const messageRenderer = useCallback(
+    (messageProps: MessageProps) => {
+      return onRenderMessage === undefined
+        ? defaultChatMessageRenderer({ ...messageProps })
+        : onRenderMessage(messageProps, defaultChatMessageRenderer);
+    },
+    [defaultChatMessageRenderer, onRenderMessage]
+  );
+
+  const messageStatusRenderer = useCallback(
+    (
+      onRenderMessageStatus:
+        | ((messageStatusIndicatorProps: MessageStatusIndicatorProps) => JSX.Element | null)
+        | undefined,
+      defaultStatusRenderer: (
+        message: ChatMessage | /* @conditional-compile-remove(data-loss-prevention) */ BlockedMessage,
+        participantCount: number,
+        readCount: number,
+        status?: MessageStatus
+      ) => JSX.Element,
+      showMessageStatus?: boolean,
+      participantCount?: number,
+      readCount?: number
+    ) => {
+      return showMessageStatus && statusToRender ? (
+        onRenderMessageStatus ? (
+          onRenderMessageStatus({ status: message.status })
+        ) : (
+          defaultStatusRenderer(message, participantCount ?? 0, readCount ?? 0, message.status)
+        )
+      ) : (
+        <div className={mergeStyles(noMessageStatusStyle)} />
+      );
+    },
+    [message, statusToRender]
+  );
+
+  const shouldShowAvatar = useMemo(() => {
+    return message.attached === 'top' || message.attached === false;
+  }, [message.attached]);
+
+  const attached = useMemo(() => {
+    return shouldShowAvatar ? 'top' : 'center';
+  }, [shouldShowAvatar]);
+
+  // Fluent UI message components are used here as for default message renderer,
+  // timestamp and author name should be shown but they aren't shown for custom renderer.
+  // More investigations are needed to check if this can be simplified with states.
+  // Status and avatar should be shown for both custom and default renderers.
+  if (message.mine === true) {
+    return (
+      <div>
+        <FluentChatMyMessage
+          attached={attached}
+          root={{
```

Comment:
Just a note that these objects are recreated each render pass and aren't memozied. Consider memoizing

Reply:
Unless v9 handles this much better than v8!

Reply:
We don't update it too often because of how it's used in MessageThread (React.memo) but I'll add useMemo here too